### DISCUSSION TITLE
How to grant access to the database tables using policies?

### DISCUSSION THREAD

**Discussion by fidel-perez** [supabase/supabase#2830] [Questions] [answered]
Hello. After watching the tutorials and trying all that I could think of, I still cant get access from my `authenticated` user to the tables using policies.
If I grant the `authenticated` role in postgreSql the permissions, he can see the tables.
But if I disable those grants in the role, and try to configure them as a policy, he wont be able to access:

Running this Sql:
```
CREATE POLICY "permissions_to_everyone" ON public.food_and_recipe_nutrients_history FOR ALL USING (true);
CREATE POLICY "permissions_to_everyone" ON public.food_categories FOR ALL USING (true);
CREATE POLICY "permissions_to_everyone" ON public.food_prep_status FOR ALL USING (true);
CREATE POLICY "permissions_to_everyone" ON public.foods FOR ALL USING (true);
CREATE POLICY "permissions_to_everyone" ON public.orders FOR ALL USING (true);
CREATE POLICY "permissions_to_everyone" ON public.stock_data FOR ALL USING (true);
CREATE POLICY "permissions_to_everyone" ON public.stock_transactions FOR ALL USING (true);
CREATE POLICY "permissions_to_everyone" ON public.store_articles FOR ALL USING (true);
```

I still get this response:

```{"hint":null,"details":null,"code":"42501","message":"permission denied for table foods"}```

Shouldn't the policies be granting all permissions to all users?

All the tables look like this after running the SQL mentioned above:

![image](https://user-images.githubusercontent.com/4046326/129354771-c63010bf-12f7-4ef2-8676-efe6c99d0273.png)

Just in case this is relevant, I also confirmed the user is properly authenticated:

![image](https://user-images.githubusercontent.com/4046326/129354361-3148b90b-15df-4c12-ae38-de28851aca2f.png)

Thank you!

**Comment by soedirgo** [comment, marked as answer]
Hey @Armagedoom, grants are separate from RLS, so if you revoked the grants from the role, it won't work. You still have to grant the permissions to `authenticated`. For sanity check, you can try creating a policy with `USING (false)` to see if the RLS works (your user should get an empty list).

**Reply by fidel-perez** [reply]
Thank you!

Knowing this, everything worked inmediatelly. 

Maybe its too obvious or its something that must be known by someone already using supabase, but if you dont think so, maybe this information belongs in the documentation.
